### PR TITLE
i3status-rust: new 0.14.2

### PR DESCRIPTION
**Pull request by peeweep**
Topic Description
-----------------

i3status-rust, a i3 status bar written in Rust.

Package(s) Affected
-------------------

i3status-rust

Security Update?
----------------

no

Architectural Progress
----------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

----

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by MingcongBai** [comment]
@peeweep Progress checklist here are used to indicate that packages are either:

- (Pre-Merge) Packaged and uploaded to its respective topic repository.
- (Post-Merge) Packaged and uploaded to the stable repository.

Since you don't yet have access to our repository server as a packager, I will review and package for you. After your first contribution you will be granted access.

File: extra-wm/i3status-rust/spec [hunk 1]
```diff
@@ -0,0 +1,3 @@
+VER=0.14.2
+SRCTBL="https://github.com/greshake/i3status-rust/archive/v$VER/i3status-rust-$VER.tar.gz"
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/spec]: Please use the new [ACBS specification file format](https://wiki.aosc.io/developer/packaging/acbs/spec-format/#spec).

File: extra-wm/i3status-rust/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,7 @@
+cargo build --release --locked
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/build]: Please annotate your build script with `abinfo`, as required by the [AOSC OS Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/#scripting).

File: extra-wm/i3status-rust/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+install -Dm755 -t "${PKGDIR}"/usr/bin target/release/i3status-rs
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/build]: Please use verbosity for all commands, as required by the [AOSC OS Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/#scripting).

File: extra-wm/i3status-rust/autobuild/build [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+install -Dm644 -t "${PKGDIR}"/usr/share/man/man1 man/i3status-rs.1
+
+for example in example_config.toml example_icon.toml example_theme.toml; do
+  install -Dm644 -t "${PKGDIR}"/usr/share/doc/${pkgname}/examples/ ${example}
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/build]: Same as above.

File: extra-wm/i3status-rust/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=i3status-rust
+PKGSEC=x11
+PKGDEP="pulseaudio"
+BUILDDEP="rustc"
+PKGDES="Very resourcefriendly and feature-rich replacement for i3status, written in pure Rust"
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/defines]: Please use descriptive and neutral remarks in `PKGDES=`, here's an acceptable example:

```
PKGDES="A Rust re-implementation of i3status"
```

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-wm/i3status-rust/autobuild/build (hunk 3 above)
Comment (peeweep) [reply extra-wm/i3status-rust/autobuild/build]: What does "verbosity" mean? Can I use `abinfo` ?

File: extra-wm/i3status-rust/autobuild/build [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+abinfo "Installing binary"
+install -Dm755 -v -t "${PKGDIR}"/usr/bin target/release/i3status-rs
+
+abinfo "Installing man page"
+install -Dm644 -v -t "${PKGDIR}"/usr/share/man/man1 man/i3status-rs.1
+
+abinfo "Installing example configuration file"
+for example in example_config.toml example_icon.toml example_theme.toml; do
+  install -Dm644 -v -t "${PKGDIR}"/usr/share/doc/"${PKGNAME}"/examples/ "${example}"
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/build]: Please use absolute path for all file references.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Comment by MingcongBai** [comment]
Great work! We will now build and test the package. Would you mind describing a test case and screenshot of it running?

**Comment by peeweep** [comment]
image

~/.config/i3/status.toml
``` toml
theme = "slick"

[icons]
name = "awesome"

[[block]]
block = "weather"
format = "{weather} ({location}) {temp}°"
service = { name = "openweathermap", api_key = "42a1771a0b787bf12e734ada0cfc80cb", city_id = "1796236", units = "metric" }
interval = 600

[[block]]
block = "memory"
format_mem = "{Mum}MB/{MTm}MB({Mup}%)"
icons = true
interval = 5

[[block]]
block = "sound"
driver = "auto"

[[block]]
block = "time"
interval = 1 
format = "%Y-%m-%d %R"
```

File: extra-wm/i3status-rust/autobuild/build [hunk 7]
```diff
@@ -0,0 +1,13 @@
+abinfo "Building binary"
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/build]: Please cut all portion before man page installation, and rename this file `beyond`.

File: extra-wm/i3status-rust/autobuild/build [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+cargo build --release --locked
+
+abinfo "Installing binary"
+install -Dm755 -v -t "${PKGDIR}"/usr/bin target/release/i3status-rs
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/build]: Please use absolute path.

File: extra-wm/i3status-rust/autobuild/build [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+abinfo "Installing man page"
+install -Dm644 -v -t "${PKGDIR}"/usr/share/man/man1 man/i3status-rs.1
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/build]: Please use absolute path.

File: extra-wm/i3status-rust/autobuild/defines [hunk 10]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=i3status-rust
+PKGSEC=x11
+PKGDEP="pulseaudio"
+BUILDDEP="rustc"
+PKGDES="A Rust re-implementation of i3status"
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/defines]: Please append the following...

```
ABTYPE=rust
USECLANG=1
```

To make use of Autobuild3’s standard build routine for Rust sources. `USECLANG=1` is used here for Rust LTO.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-wm/i3status-rust/autobuild/defines [hunk 11]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=i3status-rust
+PKGSEC=x11
+PKGDEP="pulseaudio"
+BUILDDEP="rustc"
```
Comment (MingcongBai) [inline extra-wm/i3status-rust/autobuild/defines]: Please add `llvm` to `BUILDDEP=`.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by peeweep** [comment]
~/.config/i3/config
``` config
# Start i3bar to display a workspace bar (plus the system information i3status
# finds out, if available)
bar {
        font pango:DejaVuSansMono Nerd Font Book 10 
        position top
        status_command i3status-rs ~/.config/i3/status.toml
        tray_output none
        colors {
                background #424242
        }
}
```

**Pull request closed without merging**